### ISSUE TITLE
"Run File as Script" Issue

### ISSUE THREAD

**Issue by PokeMageTech** [pyzo/pyzo#761]
When attempting to run a file as a script, if IPython is enabled for the current shell configuration, it fails to restart the shell.

**Comment by PokeMageTech** [comment]
> > When attempting to run a file as a script, if IPython is enabled for the current shell configuration, it fails to restart the shell.
> 
> You mean that it works, but the shell is not restarted beforehand? I can't remember whether this is intentional.

Yes - UNLESS the shell is dead. Then trying to run it as a script fails - you have to restart the shell yourself first.

And maybe it is intentional. I was using Pyzo2015a for a long time, so maybe it was an intentional design choice they made, like removing the ability to kill the shell as easily as you could interrupt, clear, restart, or close it. (Or really kill the shell in any way other than normal Python approaches from within the whatever the shell is executing.)

But even if it is intentional: having it be configurable would be nice. I write plenty of Cython, and it would be nice if I could just do ctrl-F5 to restart the shell and re-run the Python code that compiles the Cython module then imports it. (The shell must be killed, because the .pyd (.so on Linux) file, which is basically a Dynamically Linked Library, must unloaded before the Cython compiler -> C/C++ compiler pipeline can actually output the final, compiled file.) Now, I'm just one person, so if they don't want to change it back or add a config option or whatever for it, that's fine. Still, though...

**Comment by almarklein** [comment]
> The issues above are if IPython is in enabled in the settings for that shell. If it is not, then running file as script works...]... as I expect [...]

yeah, I would expect this to be consistent indeed.

> Now, I'm just one person, so if they don't want to change it back or add a config option or whatever for it, that's fine. 

I'm also just one person, with never enough time to work on Pyzo as I want ...

**Comment by bdieterm** [comment]
The option to normally restart the shell even for IPython could be easily implemented with a flag in the shell configuration.
And this flag could simply be just a line like `PYZO_RESTART_IPYTHON_ON_RUNSCRIPT=1` in the environ field of the shell configuration.

The following code
https://github.com/pyzo/pyzo/blob/d3c15d828a4efc9c221ee43bafc47cf873a6aeeb/pyzo/core/menu.py#L2423-L2432
would only need some additional condition like `if "PYZO_RESTART_IPYTHON_ON_RUNSCRIPT=1" in sh._startup_info["environ"].splitlines()` to check if the shell should be restarted like non-IPython shells when running files as scripts.

If this is still needed after 3 years, just reopen this case or create a new one.
